cql3/statements/select_statement.cc

```diff
@@ -1320,6 +1325,135 @@ indexed_table_select_statement::find_index_clustering_rows(service::storage_prox
     });
 }
 
+
+::shared_ptr<cql3::statements::select_statement> forwarding_select_statement::prepare(
```

Comment:
I think the name "forwarding" shows its weakness here. The word "forwarding" evokes a message being forwarded to another server that will completely handle the request. Here we're subcontracting the work to many other servers in parallel, and we're responsible for some of the work too (final aggeregation). This one isn't more or less forwarding than a regular select_statement than can be sent to a replica to be executed.

I'd call it parallelized_select_statement, but maybe there are better names.